src/does-match.js

```diff
@@ -37,45 +37,108 @@
       return string;
     }
 
-    var replaced = '';
-    for (var i = 0, size = string.length; i < size; i++) {
-      replaced += replaceDiacritics(string[i]);
-    }
+    var replaced = string.split('').map(replaceDiacritics);
 
-    return replaced;
+    return replaced.join('');
```

Comment:
Dont think this is a problem at all. See, when your application/page is loaded, the vast amount of processing goes into the SQL queries, backend, rendering or animating your page. A very low portion of the chunk that goes with your javascript is about joining/spliting arrays, or any functions like these.

This way lets you use a more fluent writing, making it easier for people to read and contribute, besides keeping your code simpler.

If you have a performance problem with your application, it likely be with your SQL queries, image sizes, css animations, or even js loops iterating over and over, but I do not think it will be about a join, split, indexOf functions of any sort. I dont event think it'd be necessary to have benchmarks...